Source/JavaScriptCore/jit/JITPlan.cpp

```diff
@@ -155,7 +163,7 @@ bool JITPlan::reportCompileTimes() const
 
 void JITPlan::compileInThread(JITWorklistThread* thread)
 {
-    m_thread = thread;
+    SetForScope threadScope(m_thread, thread);
```

Comment:
Added this change to fix the workers/bomb.html crash. What was happening was plan->m_thread was being left to the compiler thread the plan was compiled on even after the plan finished compiling and in the 'Ready' state. Then, the test causes the VM to be destructed, which cancels all plans that aren't actively being compiled in the worklist (and waits for the compiling ones to complete). When canceling the plan with a dangling reference, the thread that it was pointing to happened to be inside the B3::generate() safe point but for a different plan, causing the new RELEASE_ASSERT() in Plan::cancel() to trigger incorrectly.